docs_src/microservices/device/V3Migration.md

```diff
@@ -8,29 +8,152 @@ See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details app
 
 ### Device Files
 
-TBD
+1. Change device definition file in the Device Service to YAML format.
+2. Remove `LastConnected` and `LastReported` configs.
+3. ProtocolProperties now supports typed values.
+
+    !!! example "ProtocolProperty with typed values"
+    
+        ```yaml
+        protocols:
+          other:
+            Address: simple01
+            Port: 300
+        ```
 
 ### Device Profile Files
 
-TBD
+1. Add `optional` field in ResourceProperties to allow any additional or customized data.
+2. Change the data type of `mask`, `shift`, `scale`, `base`, `offset`, `maximum` and `minimum` from string to number in ResourceProperties.
+
 
 ### Provision Watcher files
 
-TBD
+1. Allow empty string profile name when adding or updating the ProvisionWatcher.
+2. The ProvisionWatcher DTO is restructured by moving the Device related fields into a new object field, `DiscoveredDevice`; such as `profileName`, Device `adminState`, and `autoEvents`.
+3. Allow to define additional or customized data by utilizing the `properties` field in the `DiscoveredDevice` object.
+4. ProvisionWatcher contains its own `adminState` now. The Device `adminState` is moved into the `DiscoveredDevice` object.
+5. ProvisionWatcher can now be added during device service startup by loading the definition files from the `ProvisionWatchersDir` configuration.
+
+    !!! example "Example Configuration"
+    
+        ```yaml
+        Device:
+            ProvisionWatchersDir: ./res/provisionwatchers
+        ```
+
+6. ProvisionWatcher definition file is in YAML format.
+
+    !!! example "Pre-defined ProvisionWatcher"
+
+        ```yaml
+        name: Simple-Provision-Watcher
+        serviceName: device-simple
+        labels:
+          - simple
+        identifiers:
+          Address: simple[0-9]+
+          Port: 3[0-9]{2}
+        blockingIdentifiers:
+          Port:
+            - 397
+            - 398
+            - 399
+        adminState: UNLOCKED
+        discoveredDevice:
+          profileName: Simple-Device
+          adminState: UNLOCKED
+          autoEvents:
+            - interval: 15s
+              sourceName: SwitchButton
+          properties:
+            testPropertyA: weather
+            testPropertyB: meter
+        ```
 
 ### Other
 
-TBD
+1. The `-c/--confdir` command line option is replaced by `-cd/--configDir`.
+2. The `-f/--file` command line option is replaced by `-cf/--configFile`.
```

Comment:
This is part of common migration. Don't duplicate here

Reply:
removed